FridaTests/MacOsTargetTests.swift

```diff
@@ -0,0 +1,162 @@
+import XCTest
+@testable import Frida
+
+fileprivate enum TestScriptMessage {
+    case regular(type: String, payload: String)
+    case other(message: Any, data: Data?)
+}
+
+fileprivate class TestScriptDelegate: NSObject, ScriptDelegate {
+    var onMessage: ((TestScriptMessage) -> Void)?
+    
+    @objc func script(_ script: Script, didReceiveMessage message: Any, withData data: Data?) {
+        guard let dict = message as? [String: String] else {
+            onMessage?(.other(message: message, data: data))
+            return
+        }
+        
+        let typeMaybe = dict["type"]
```

Comment:
The `type` field is guaranteed to be there. (The `payload` property might be missing though, in case the user did `send(undefined)`, which is a bit unfortunate.)